acb/utf.py

```diff
@@ -161,7 +161,18 @@ def string0(self, at=None):
             break
         string = JOIN_BYTE_ARRAY(sr)
         self.f.seek(bk + len(string) + 1)
-        return string.decode("sjis")
+        return try_decode(string)
+
+
+def try_decode(text):
+    charsets = ['utf8', 'sjis']
```

Comment:
Can you change this to try sjis first? This is to avoid changing the behaviour for ASCII filenames with `\` in them (which would be decoded to `¥` with sjis, and `\` with utf8).